pkg/provider/branch/provider.go

```diff
@@ -469,7 +469,7 @@ func (b *branchENIProvider) IsInstanceSupported(instance ec2.EC2Instance) bool {
 	limits, found := vpc.Limits[instance.Type()]
 	supported := found && instance.Os() == config.OSLinux && limits.IsTrunkingCompatible
 
-	if !supported {
+	if !supported && instance.Os() == config.OSLinux {
```

Comment:
Nit: for readability, should we have the logic like

```
// SGPP is not supported on Windows nodes
if instance.Os() == config.OSWindows{
  return false
} else {
  limits, found := vpc.Limits[instance.Type()]
  if  !(found && limits.IsTrunkingCompatible) {
    (Linux node- raise event)
    return false
  }
}
return true
```

Reply:
I agree. Returning Windows as false earlier makes sense to me. Will make the change.